src/interfaces/domain/customer/repository/repository.ts

```diff
@@ -0,0 +1,3 @@
+export default interface IRepository<T> {
```

Comment:
Esse IRepository será usado em outros contextos? Talvez seja bom tirar ele e customer caso for.

Reply:
realmente, corrigido